packages/core/src/__tests__/cogniteClient.unit.spec.ts

```diff
@@ -131,6 +131,102 @@ describe('CogniteClient', () => {
 
         expect(getToken).toHaveBeenCalledTimes(1);
       });
+
+      test('getToken should be called once for parallel 401s', async () => {
+        nock(mockBaseUrl).get('/test').thrice().reply(401);
+        nock(mockBaseUrl).get('/test').thrice().reply(200);
+
+        const mockGetToken = jest.fn(
+          () =>
+            new Promise<string>((resolve) => {
+              setTimeout(() => {
+                resolve('test-token');
+              }, 100);
+            })
+        );
+
+        const client = new BaseCogniteClient({
+          project,
+          appId: 'unit-test',
+          baseUrl: mockBaseUrl,
+          apiKeyMode: true,
+          getToken: mockGetToken,
+        });
+
+        await Promise.all([
+          client.get('/test'),
+          client.get('/test'),
+          client.get('/test'),
+        ]);
+
+        expect(mockGetToken).toBeCalledTimes(1);
+      });
+
+      test('getToken should be called once for parallel 401s with different response times', async () => {
+        nock(mockBaseUrl).get('/test').twice().reply(401);
+        nock(mockBaseUrl).get('/test-with-delay').delay(200).reply(401);
+        nock(mockBaseUrl).get('/test').twice().reply(200);
+        nock(mockBaseUrl).get('/test-with-delay').reply(200);
+
+        const mockGetToken = jest.fn(
+          () =>
+            new Promise<string>((resolve) => {
+              setTimeout(() => {
+                resolve('test-token');
+              }, 100);
+            })
+        );
+
+        const client = new BaseCogniteClient({
+          project,
+          appId: 'unit-test',
+          baseUrl: mockBaseUrl,
+          apiKeyMode: true,
+          getToken: mockGetToken,
+        });
+
+        await Promise.all([
+          client.get('/test'),
+          client.get('/test'),
+          client.get('/test-with-delay'),
+        ]);
+
+        expect(mockGetToken).toBeCalledTimes(1);
+      });
+
+      test('getToken should be called more than once for sequential 401s', async () => {
+        nock(mockBaseUrl).get('/test').thrice().reply(401);
+        nock(mockBaseUrl).get('/test').thrice().reply(200);
+        nock(mockBaseUrl).get('/test').reply(401);
+        nock(mockBaseUrl).get('/test').reply(200);
+
+        const mockGetToken = jest.fn(
+          () =>
+            new Promise<string>((resolve) => {
+              setTimeout(() => {
+                resolve(`test-token-${Math.floor(Math.random() * 1000)}`);
```

Comment:
Can we make this part deterministic? Ex. `test-token-${tokenCount++}`.
And utilize the `sleepPromise` function.